defines.mk

```diff
@@ -0,0 +1,25 @@
+include ${FLEXPRET_ROOT_DIR}/config.mk
+
+DEFINES :=
```

Comment:
The DEFINES variable? I meant to use it as a variable that just contains -D flags for defining macros through the makefile. The advantage of doing this is that you don't need to worry about whether it is defined in the correct place all the time. In the case of including the reent stuff from newlib, it becomes much simpler because it expects you to have some macros defined before the .h file is included. In this case it is very helpful to have the macros defined for the entire compilation to avoid getting bugs from the macros not being defined in one place.

Also it does the same thing for printf configuration now